source/mod_pythia.f90

```diff
@@ -0,0 +1,398 @@
+
+  case("PROCESS")
+    if(nSplit /= 2 .and. nSplit /= 3) then
+      write(lout,"(a,i0)") "PYTHIA> ERROR Keyword PROCESS expected 1 or 2 arguments, got ",(nSplit-1)
+      iErr = .true.
+      return
+    end if
+    select case(lnSplit(2))
+    case("EL","ELASTIC")
+      pythia_useElastic = .true.
+      write(lout,"(a)") "PYTHIA> Elastic scattering enabled"
+      if(nSplit > 2) then
+        call chr_cast(lnSplit(3),pythia_csElastic,iErr)
+        write(lout,"(a,f13.8,a)") "PYTHIA> Elastic cross section set to ",pythia_csElastic," mb"
+      end if
+    case("SD","SINGLEDIFFRACTIVE")
+      pythia_useSDiffractive = .true.
+      write(lout,"(a)") "PYTHIA> Single diffractive scattering enabled"
+      if(nSplit > 2) then
+        call chr_cast(lnSplit(3),pythia_csSDiffractive,iErr)
+        write(lout,"(a,f13.8,a)") "PYTHIA> Single diffractive cross section set to ",pythia_csSDiffractive," mb"
+      end if
+    case("DD","DOUBLEDIFFRACTIVE")
+      pythia_useDDiffractive = .true.
+      write(lout,"(a)") "PYTHIA> Double diffractive scattering enabled"
+      if(nSplit > 2) then
+        call chr_cast(lnSplit(3),pythia_csDDiffractive,iErr)
+        write(lout,"(a,f13.8,a)") "PYTHIA> Double diffractive cross section set to ",pythia_csDDiffractive," mb"
+      end if
+    case("CD","CENTRALDIFFRACTIVE")
+      pythia_useCDiffractive = .true.
+      write(lout,"(a)") "PYTHIA> Central diffractive scattering enabled"
+      if(nSplit > 2) then
+        call chr_cast(lnSplit(3),pythia_csCDiffractive,iErr)
+        write(lout,"(a,f13.8,a)") "PYTHIA> Central diffractive cross section set to ",pythia_csCDiffractive," mb"
+      end if
+    case("ND","NONDIFFRACTIVE")
+      pythia_useNDiffractive = .true.
+      write(lout,"(a)") "PYTHIA> Non-diffractive scattering enabled"
+      if(nSplit > 2) then
+        call chr_cast(lnSplit(3),pythia_csNDiffractive,iErr)
+        write(lout,"(a,f13.8,a)") "PYTHIA> Non-diffractive cross section set to ",pythia_csNDiffractive," mb"
+      end if
+    case default
+      write(lout,"(a)") "PYTHIA> ERROR Unknown or unsupported scattering process'"//trim(lnSplit(2))//"'"
+      iErr = .true.
+      return
+    end select
+
+  case("COULOMB")
+    if(nSplit /= 2 .and. nSplit /= 3) then
+      write(lout,"(a,i0)") "PYTHIA> ERROR Keyword COULOMB expected 1 or 2 arguments, got ",(nSplit-1)
+      iErr = .true.
+      return
+    end if
+
+    call chr_cast(lnSplit(2),pythia_useCoulomb,iErr)
+    if(nSplit == 3) then
+      call chr_cast(lnSplit(3),pythia_elasticTMin,iErr)
+      if(pythia_elasticTMin < 1.0e-10 .or. pythia_elasticTMin > 1.0e-3) then
+        write(lout,"(a)") "PYTHIA> ERROR Range for COULOMB TMIN is 1e-10 to 1e-3."
+        iErr = .true.
+        return
+      end if
+    end if
+
+  case("SPECIES")
+    if(nSplit /= 3) then
+      write(lout,"(a,i0)") "PYTHIA> ERROR Keyword BEAM expected 2 arguments, got ",(nSplit-1)
```

Comment:
BEAM or SPECIES?

Reply:
SPECIES. Just missed it after copy/paste.